core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -284,20 +284,22 @@ oppia.controller('StateResponses', [
       trainingDataService.initializeTrainingData(
         explorationId, currentStateName);
     };
+
     $scope.suppressDefaultAnswerGroupWarnings = function() {
       var interactionId = $scope.getCurrentInteractionId();
+      var answerGroups = responsesService.getAnswerGroups();
+      var handledAnswersArray = [];
+      var choiceIndices = [];
+      var answerChoices = [];
       if (interactionId === 'MultipleChoiceInput') {
-        var answerGroups = responsesService.getAnswerGroups();
+        var numChoices = $scope.getAnswerChoices().length;
```

Comment:
Oh, I think I know what the error might be -- not every interaction has a getAnswerChoices() associated with it, so moving this outside would actually be incorrect. Thanks for the clarification; I think it's fine to leave it as-is.